### ISSUE TITLE
helm version times out in Concourse in Kubernetes

### ISSUE THREAD

**Issue by andyhd** [linkyard/concourse-helm-resource#45]
We have Concourse deployed in our Kubernetes cluster and when the concourse-helm-resource sets up helm, it hangs when running `helm version`:
```
Initializing kubectl...
Cluster "default" set.
User "admin" set.
Context "default" created.
Switched to context "default".
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:21:50Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.9", GitCommit:"3fb1aafdafa3d33bc698930095db1e56c0f76452", GitTreeState:"clean", BuildDate:"2018-03-12T16:13:32Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Initializing helm...
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Kubernetes: &version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.9", GitCommit:"3fb1aafdafa3d33bc698930095db1e56c0f76452", GitTreeState:"clean", BuildDate:"2018-03-12T16:13:32Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
[debug] context deadline exceeded
Error: cannot connect to Tiller
```

We tracked this down to [`setupConnection()` in `helm.go`](https://github.com/kubernetes/helm/blob/a80231648a1473929271764b920a8e346f6de844/cmd/helm/helm.go#L169), which creates a tunnel to Tiller, unless `HELM_HOST` is set.

Setting `HELM_HOST` to `tiller-deploy.kube-system:44134` solved our problem.

I have added a source configuration setting for `HELM_HOST`, so that this can be set in the pipeline, if  you would like me to make a PR.

**Comment by furuholm** [comment]
I have the same issue and I suspect we are seing this: https://github.com/kubernetes/helm/issues/2409 which should be fixed by https://github.com/kubernetes/helm/pull/3715. This has not made it into the released version of helm yet...

**Comment by furuholm** [comment]
https://github.com/kubernetes/helm/pull/3715 has been merged. Will make it into 2.9.0-RC4 which will be released tomorrow...

**Comment by furuholm** [comment]
2.9.0-RC4 has now been released https://github.com/kubernetes/helm/releases/v2.9.0-rc4

**Comment by msiegenthaler** [comment]
I’ll release a new version as soon as the final 2.9.0 is released

**Comment by furuholm** [comment]
I have verified that 2.9.0-RC4 does **not** fix the issue. Setting the helm host manually does however. @msiegenthaler: would you consider accepting the PR that was suggested by @andyhd? The helm host seems like a reasonable parameter to be able to set even if this bug did not require it.

**Comment by msiegenthaler** [comment]
yes, I'd accept a PR that sets `HELM_HOST`. Can you open that PR @andyhd ?

**Comment by andyhd** [comment]
Here it is, sorry for the delay!

**Comment by msiegenthaler** [comment]
fixed by #49
